introduce a github action for open source release

subodh1810 opened the pull request:
Issue : https://github.com/airbytehq/airbyte/issues/5459
Action to create a PR : https://github.com/marketplace/actions/create-pull-request



## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>


<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

subodh1810 commented:
TODO(Subodh) : Need to add `CLOUDREPO_USER` and `CLOUDREPO_PASSWORD` to the secrets

jrhizor commented on .github/workflows/release-airbyte-os.yml:
```diff
@@ -0,0 +1,53 @@
+        run: |
+          echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
+          echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
+      - name: Next Steps
```
Can we put these steps in the PR body instead of the Github Action? I don't think anyone will look at this usually.

jrhizor commented on tools/bin/release_version.sh:
```diff
@@ -44,15 +36,3 @@ SUB_BUILD=PLATFORM ./gradlew publish
 echo "Completed building and publishing..."
-
-echo
-echo "Changelog:"
```
Not blocking but nice to have:

It'd be cool if the changelog could also be put in the PR. I imagine you'd have to compute the changelog, save it to an output variable and then read it in the PR creation step?

tuliren replied:
This is done directly in the `pr_body.sh` now.

jrhizor commented on .github/workflows/release-airbyte-os.yml:
```diff
@@ -0,0 +1,53 @@
+      - name: Next Steps
+        run: |
+          echo "Steps After Merging PR"
+          echo "1. Switch to master"
```
```suggestion
          echo "1. Pull most recent version of master"
```

jrhizor reviewed (approved):
Looks good overall. Just minor comments.

subodh1810 commented:
@jrhizor I addressed your comments, take a look at this PR https://github.com/airbytehq/github-workflow-test-repo-base/pull/30 , this is how the output PR would look from this new workflow. Let me know if you have any more feedback

davinchia commented on .github/workflows/release-airbyte-os.yml:
```diff
@@ -0,0 +1,58 @@
+      - name: PR Details
+        run: |
+          echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
+          echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
```
nit: newline

davinchia commented on tools/bin/pr_body.sh:
```diff
@@ -0,0 +1,17 @@
+echo "Steps After Merging PR:"
+echo "1. Pull most recent version of master"
+echo "2. Run ./tools/bin/tag_version.sh"
+echo "3. Create a GitHub release with the changelog"
```
nit: newline

davinchia commented on .github/workflows/release-airbyte-os.yml:
```diff
@@ -0,0 +1,58 @@
+        env:
+          PREV_VERSION: ${{ steps.old_version.outputs.OLD_VERSION }}
+        run: |
+          chmod +x tools/bin/pr_body.sh
```
we can `chmod` the `pr_body.sh` script and check that in so we don't have to do this in the workflow everytime.

davinchia commented on .bumpversion.cfg:
```diff
@@ -1,6 +1,6 @@
 [bumpversion]
 current_version = 0.29.13-alpha
-commit = True
+commit = False
```
to make sure I understand, this is false because we are manually creating the commit later on in the workflow yes?

subodh1810 replied:
@davinchia yup..now we dont want `bumpversion` to commit changes. It would be taken care by the workflow

davinchia reviewed (approved):
Looks good!

tuliren commented:
Wow, this is fancy!
